pkg/node/node.go

```diff
@@ -92,17 +97,44 @@ func NewBee(o Options) (*Bee, error) {
 	}
 	b.p2pService = p2ps
 
+	// TODO: be more resilient on connection errors and connect in parallel
+	for _, a := range o.Bootnodes {
+		addr, err := ma.NewMultiaddr(a)
+		if err != nil {
+			return nil, fmt.Errorf("bootnode %s: %w", a, err)
+		}
+
+		overlay, err := p2ps.Connect(p2pCtx, addr)
+		if err != nil {
+			return nil, fmt.Errorf("connect to bootnode %s %s: %w", a, overlay, err)
```

Comment:
this is kind of tricky. if you can't connect to the bootnodes for some reason this does not necessarily infer that the node should be taken down.
i see a few options here, speaking purely from a usability perspective:
- if the bootnodes flag was supplied by the cli explicitly and the connection to all of them failed - then return an error (we must have this functionality for proper cluster bootstrapping with k8s i believe). you can enforce this with an `ErrorGroup`.
- if no cli flag was supplied for now we should never return an error here

Reply:
This code is copied from libp2p package that I've written earlier. I think that it is outside of the scope of this PR and there already is a comment a few lines above.

> if no cli flag was supplied for now we should never return an error here

This is true as if Bootnodes slice is empty, this code will never be executed.

This package has nothing to do with cli flags. There are only passed options to the constructor. Just a note on the scope of this package, even it is used in cmd.

Reply:
> This package has nothing to do with cli flags. There are only passed options to the constructor. Just a note on the scope of this package, even it is used in cmd.


this is why I wrote `speaking purely from a usability perspective`.

Reply:
anyway - ok. let's keep as is